### PR TITLE
feat(RHTAPREL-714): add Tekton utilities

### PR DESCRIPTION
**Pull request by davidmogar**
The utilities added in this commit come from the shared-utils repository. In addition, the RPA already references the new Pipeline struct.

Signed-off-by: David Moreno García <damoreno@redhat.com>

Depends on #310

### CODE REVIEW

**Review by mmalina — DISMISSED** [review]

File: tekton/utils/suite_test.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2022.
+Copyright 2022 Red Hat Inc.
```
Comment (johnbieren) [inline tekton/utils/suite_test.go:2]: wrong year

**Review by johnbieren — COMMENTED** [review]
This needs to be merged with e2e-tests changes and tenant config changes. It also has more things to change since you rebased

File: tekton/utils/pipeline.go [hunk 2]
```diff
@@ -26,24 +35,35 @@ type PipelineRef struct {
...
-type Param struct {
-	// Name is the name of the parameter
-	Name string `json:"name"`
-
-	// Value is the value of the parameter
-	Value string `json:"value"`
+// ParametrizedPipeline is an extension of the Pipeline struct, adding an array of parameters that will be passed to
+// the Pipeline.
+// +kubebuilder:object:generate=true
+type ParametrizedPipeline struct {
```
Comment (johnbieren) [inline tekton/utils/pipeline.go:60]: Can we call it ParameterizedPipeline? Parametrized isn't a word

File: tekton/utils/suite_test.go [hunk 3]
```diff
@@ -17,22 +17,59 @@ limitations under the License.
...
+	testEnv = &envtest.Environment{}
+
+	var err error
+	// cfg is defined in this file globally.
+	cfg, err = testEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(cfg).NotTo(BeNil())
+
+	//+kubebuilder:scaffold:scheme
+	k8sClient, err = client.New(cfg, client.Options{
```
Comment (johnbieren) [inline tekton/utils/suite_test.go:63]: what do we need this for? It seems like the tests are just manipulating structs unless I missed something

File: tekton/utils/suite_test.go (hunk 1 above)
Comment (davidmogar) [reply tekton/utils/suite_test.go]: Fixed.

File: tekton/utils/pipeline.go (hunk 2 above)
Comment (davidmogar) [reply tekton/utils/pipeline.go]: Fixed.

File: tekton/utils/suite_test.go (hunk 3 above)
Comment (davidmogar) [reply tekton/utils/suite_test.go]: Removed.

**Review by johnbieren — COMMENTED** [review]
LGTM but not marking approval to make sure this doesn't get accidentally merged

**Review by johnbieren — DISMISSED** [review]

File: go.sum [hunk 4]
```diff
@@ -294,6 +294,7 @@ github.com/redhat-appstudio/integration-service v0.0.0-20231025084434-b3f521c408
 github.com/redhat-appstudio/integration-service v0.0.0-20231025084434-b3f521c408d1/go.mod h1:bh8uJ4whJ92ef5PXH6Rfm4F8tRXkDAlSYR7Pqp1LLT4=
 github.com/redhat-appstudio/operator-toolkit v0.0.0-20231201124606-2087182322ae h1:98thWrMrr9YUNCVNGqkZxKfZkaWUmMC5dmkgAI7p3uY=
 github.com/redhat-appstudio/operator-toolkit v0.0.0-20231201124606-2087182322ae/go.mod h1:7cX2+4KGZLJ4Yoj+1v0iV5hkCGBzbSd9wkNJQjCdDJs=
+github.com/redhat-appstudio/release-service v0.0.0-20230511145849-bde1cdcbb60b h1:96jgqIR8Otx2vrLZscvNo3gFlX2d9P/mkbx5Trkm8tQ=
```
Comment (johnbieren) [inline go.sum:297]: why do we import release-service in the release-service repo?

**Review by johnbieren — APPROVED** [review]

**Pull request merged**